tests/testthat/test-interface.R

```diff
@@ -2,8 +2,23 @@ test_that("Can compile a simple model", {
   path <- mode_file("examples/logistic.cpp")
   gen <- mode(path, quiet = TRUE)
   pars <- list(r1 = 0.1, r2 = 0.2, K1 = 100, K2 = 100)
-  mod <- gen$new(pars, pi)
+  n_particles <- 10
+  mod <- gen$new(pars, pi, n_particles)
   expect_s3_class(mod, "mode")
   expect_equal(mod$time(), pi)
   expect_equal(mod$pars(), pars)
 })
+
+test_that("Returned state has correct dimensions", {
+  path <- mode_file("examples/logistic.cpp")
```

Comment:
I'll get a PR ready that adds the same caching as dust ready. That won't interfere with this PR, but it will speed up adding more tests